src/redux/books/books.js

```diff
@@ -1,35 +1,30 @@
-const ADD_BOOK = 'bookstore/books/ADD_BOOK';
-const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+import { createSlice } from '@reduxjs/toolkit';
+import { getBooks } from '../apiConnection';
 
 const initialState = {
   books: [],
+  status: null,
 };
 
-const reducer = (state = initialState, action) => {
-  switch (action.type) {
-    case ADD_BOOK:
-      return {
-        ...state,
-        books: [...state.books, action.payload],
-      };
-    case REMOVE_BOOK:
-      return {
-        ...state,
-        books: state.books.filter((book) => book.id !== action.payload),
-      };
-    default:
-      return state;
-  }
-};
-
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
-
-export const deleteBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: id,
+const reducer = createSlice({
+  name: 'books',
+  initialState,
+  extraReducers: {
+    [getBooks.pending]: (state) => ({
+      ...state,
+      status: 'loading',
+    }),
+    [getBooks.fulfilled]: (state, action) => ({
+      ...state,
+      books: action.payload,
+      status: 'success',
+    }),
+    [getBooks.rejected]: (state) => ({
+      ...state,
+      status: 'error',
+    }),
+  },
 });
 
+// export const { addBook, deleteBook } = reducer.actions;
```

Comment:
- [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.